libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationRouteProcessor.java

```diff
@@ -28,10 +34,21 @@
 
   private static final int FIRST_LEG_INDEX = 0;
   private static final int FIRST_STEP_INDEX = 0;
-
-  private RouteProgress previousRouteProgress;
-  private List<Point> stepPoints;
+  private static final int ONE_INDEX = 1;
+
+  private RouteProgress routeProgress;
+  private List<Point> currentStepPoints;
+  private List<Point> upcomingStepPoints;
+  private List<StepIntersection> currentIntersections;
+  private List<Pair<StepIntersection, Double>> currentIntersectionDistances;
+  private RouteLeg currentLeg;
+  private LegStep currentStep;
+  private LegStep upcomingStep;
+  private CurrentLegAnnotation currentLegAnnotation;
   private NavigationIndices indices;
+  private double routeDistanceRemaining;
+  private double legDistanceRemaining;
+  private double stepDistanceRemaining;
   private boolean shouldIncreaseIndex;
 
   NavigationRouteProcessor() {
```

Comment:
Although theses changes have refactored `NavigationRouteProcessor` there are still some methods kinda long. What about extracting blocks of code into well-named (based on the comments if there are any) `private` methods and simplifying conditional expressions?